src/renderer/api/kube-api.ts

```diff
@@ -232,8 +234,43 @@ export class KubeApi<T extends KubeObject = any> {
     return this.resourceVersions.get(namespace);
   }
 
-  async refreshResourceVersion(params?: { namespace: string }) {
-    return this.list(params, { limit: 1 });
+  async listMetadata({ namespace = "" } = {}): Promise<KubeJsonApiListMetadataParsed> {
+    const response = await this.rawList({ namespace }, {
+      limit: 1, // specifying a limit to get metadata.remainingItemCount in response
+      resourceVersion: this.getResourceVersion(namespace) ?? "1",
+      resourceVersionMatch: "NotOlderThan",
+    });
+
+    const { remainingItemCount, ...metadata } = this.parseMetadata(namespace, response.metadata);
+
+    return {
+      ...metadata,
+      itemsCount: remainingItemCount + 1, // +1 from limit=1
```

Comment:
`remainingItemCount` is interesting idea but count is not updated "live". Also I'm not sure if it's available on older kube versions.

Reply:
in addition we could calculate actual loaded items in store per `namespace` for better results and/or refresh this a bit more often (it's lightweight request)